mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -284,6 +284,8 @@ class DenseMatrix(
   override def apply(i: Int, j: Int): Double = values(index(i, j))
 
   private[mllib] def index(i: Int, j: Int): Int = {
+    require(i < numRows && i >=0, s"Expected 0 <= i < $numRows, got $i")
```

Comment:
Nit: space after `>=`, and yes I think these can be combined into one check each. If there's an error I think it's helpful to log both i and j anyway.